csrc/type.h

```diff
@@ -995,7 +995,7 @@ inline PolymorphicValue castToDtype(
 
 // Convert an enum to its underlying type.
 template <typename E>
-constexpr auto toUnderlying(E e) noexcept {
```

Comment:
The new name is more descriptive. Note however that originally I named it after the C++23 feature https://en.cppreference.com/w/cpp/utility/to_underlying. Because of this it might be best to rename it `std::to_underlying` and place it in a C++23 directory like we are temporarily doing with C++20 features. Paging @zasdfgbnm for his opinion on shadowing C++23 functionality.

Reply:
I found a name conflict with `toUnderlying` with pytorch. It is defined for the `c10::ScalarType`. https://github.com/pytorch/pytorch/blob/main/c10/core/ScalarType.h#L363

Reply:
I switched to back to the original name but specified the `nvfuser` namespace to avoid conflicts.